apps/sms/services/js/messaging/messaging_service.js

```diff
@@ -65,6 +70,20 @@ var MessagingService = {
     this.initService();
   },
 
+  register(appInstanceId) {
+    var mobileMessageClient = MozMobileMessageClient.forApp(appInstanceId);
+
+    EVENTS.forEach((event) => {
+      mobileMessageClient.on(event, (data) => {
+        // Workaround to avoid the duplicated events by inserting appInstanceId
+        // into event name in bridge mixin.
```

Comment:
If there will be one lowered window for all the instances' iframe, then yes. But it seem that lowered window opened in different instance will still duplicated per my experiment.